types/src/interpolation.c

```diff
@@ -92,18 +104,43 @@ static rage_FrameSeries * as_frameseries(
             .mode = p->mode
         };
     }
-    return fs;
 }
 
-static void frameseries_free(
-        rage_TupleDef const * const tuple_def, rage_FrameSeries * points) {
-    rage_countdown_free(points->c);
-    for (uint32_t i = 0; i < points->len; i++) {
-        rage_tuple_free(tuple_def, points->items[i].value);
-    }
-    points->len = 0;
-    free(points->items);
-    free(points);
+typedef struct {
+    rage_Queue * q;
+    rage_QueueItem * qi;
+    rage_Event * event;
+    rage_TupleDef const * tuple_def;
+    rage_FrameSeries fs;
+    sem_t * change_sem;
+} rage_FrameSeriesChangedInfo;
+
+static void frameseries_done_with(void * vp) {
+    rage_FrameSeriesChangedInfo * fsi = vp;
+    rage_queue_put_block(fsi->q, fsi->qi);
+}
+
+static void frameseries_change_event_free(void * fscv) {
```

Comment:
`fscv` is the same type of thing as `vp` above? Perhaps they should be named the same...